src/Servers/Kestrel/Transport.Sockets/src/SocketTransportOptions.cs

```diff
@@ -12,24 +10,8 @@ namespace Microsoft.AspNetCore.Server.Kestrel.Transport.Sockets
     /// <summary>
     /// Options for socket based transports.
     /// </summary>
-    public class SocketTransportOptions
+    public class SocketTransportOptions : SocketConnectionFactoryOptions
     {
-        /// <summary>
-        /// The number of I/O queues used to process requests. Set to 0 to directly schedule I/O to the ThreadPool.
-        /// </summary>
-        /// <remarks>
-        /// Defaults to <see cref="Environment.ProcessorCount" /> rounded down and clamped between 1 and 16.
-        /// </remarks>
-        public int IOQueueCount { get; set; } = Math.Min(Environment.ProcessorCount, 16);
```

Comment:
I wonder if this is a breaking change?

Reply:
Code check said no as well: Detected modification to baseline API files. PublicAPI.Shipped.txt files should only be updated after a major release. See /docs/APIBaselines.md for more information.

So back this out and just duplicate this in a new options class?

Reply:
Yep

Reply:
I don't think introducing a new base class by itself is a breaking change, but I'm pretty sure moving properties down to the new base class is. https://docs.microsoft.com/en-us/dotnet/core/compatibility/ says "Introducing a new base class" "REQUIRES JUDGMENT".

Reply:
Well I reverted that and just added an internal copy constructor to flow things to the 'base' ContextFactory options

Reply:
So we could do this in a non breaking way by making the properties virtual and override them to call base or something?

Reply:
What you have is fine. Leave them separate.